src/Compilers/CSharp/Test/Syntax/IncrementalParsing/IncrementalParsingTests.cs

```diff
@@ -2341,6 +2341,51 @@ public void InsertOpenBraceBeforeCodes()
             CompareIncToFullParseErrors(reparsedTree, parsedTree);
         }
 
+        [WorkItem(6676, "https://github.com/dotnet/roslyn/issues/6676")]
+        [Fact]
+        public void InsertExpressionStatementWithoutSemicolonBefore()
+        {
+            SourceText oldText = SourceText.From(@"
+System.Console.WriteLine(true)\n
+");
+            var startTree = SyntaxFactory.ParseSyntaxTree(oldText);
+
+            var newText = oldText.WithChanges(new TextChange(new TextSpan(0, 0), "System.Console.WriteLine(false)\n"));
+            var reparsedTree = startTree.WithChangedText(newText);
+            var parsedTree = SyntaxFactory.ParseSyntaxTree(newText);
```

Comment:
It might be helpful to assert errors from the original tree as well. This will help to verify that we are testing what we intend to test.

---
In reply to: [105333422](https://github.com/dotnet/roslyn/pull/17668#discussion_r105333422) [](ancestors = 105333422)